napari/_vispy/layers/base.py

```diff
@@ -242,7 +242,7 @@ def _on_matrix_change(self):
                 ::-1
             ]
             - translate
-        )
+        ) / self.layer.scale[self.layer._slice_input.displayed][::-1]
```

Comment:
suggestion: factor out `self.layer._slice_input.displayed[::-1]` to `displayed_vispy` or something like that, for readability.

Do you have a code snippet to test that this works and doesn't break?

Reply:
The code snippet is in a linked issue.

I could make a followup PR with improved readability, but this change is done this way for easy cherry-pick.